client/signup/steps/user/index.jsx

```diff
@@ -167,6 +167,10 @@ export class UserStep extends Component {
 
 	getLoginUrl() {
 		const { oauth2Client, wccomFrom, isReskinned, sectionName, from, locale } = this.props;
+		const emailAddress =
+			typeof this.props?.step?.form?.email === 'object'
+				? this.props?.step?.form?.email?.value
+				: this.props?.step?.form?.email;
```

Comment:
Nit: Perhaps we can make it slightly shorter
```suggestion
		const { oauth2Client, wccomFrom, isReskinned, sectionName, from, locale, step } = this.props;
		const emailAddress = step?.form?.email?.value ?? step?.form?.email;
```